src/Psalm/Internal/Type/SimpleAssertionReconciler.php

```diff
@@ -2507,4 +2517,50 @@ function (Type\Union $t): bool {
             ? Type::getEmpty()
             : Type::getMixed();
     }
+
+    /**
+     * @param   0|1|2    $failed_reconciliation
+     */
+    private static function reconcileClassConstant(
+        Codebase $codebase,
+        string $class_constant_expression,
+        int &$failed_reconciliation
+    ) : Union {
+        if (strpos($class_constant_expression, '::') === false) {
+            $failed_reconciliation = 2;
```

Comment:
TBH: I have no idea for what this is for

If someone gives some clarification, some `Reconciler::RECONCILIATION_*` constants might make sense. Would be happy to implement these.

Reply:
If I'm not mistaken, 0 means no issue, 1 means the reconciliation did not remove any existing type (therefore, it was redundant, like checking is_string when we already have a TString), 2 means the resulting type is empty so there's no possible issue (like checking is_string when we have a TInt)

Reply:
Added constants for these and changed code